### ISSUE TITLE
HTR text overlay unreadable for dense documents

### ISSUE THREAD

**Issue by benwbrum** [benwbrum/fromthepage#3976]
We've started experimenting with several different types of documents using the AI Assist feature, and discovered that the font size on the text overlay probably needs to scale with the number of lines on the image.  In particular, images with dense writing have text displayed in a font size which is too large for the user to read.

See [this example](https://fromthepage.com/ai/htr-sandbox/manuscript-letter-listing-agents-of-the-adelsverein-reproduced-by-offset/display/33968559)

![Screenshot from 2024-03-05 06-58-03](https://github.com/benwbrum/fromthepage/assets/199961/c98de681-e33d-4f5f-84f1-e1ed30a80b1f)
![Screenshot from 2024-03-05 06-57-45](https://github.com/benwbrum/fromthepage/assets/199961/3e4c2b6d-a8c4-4cad-b3ea-c52925c51dab)

**Comment by rsimon** [comment]
Hi @benwbrum,

That's a challenging or a trivial issue, depending on how you look at it ;-)

We are currently using the `fixedPageSize` mode for the text overlay. That means the font gets a fixed size relative to the page, and gets larger/smaller in sync with the current zoom level. The base font size is determined by CSS and (currently) hard-wired (I should probably fix that...). As a quick fix, you could simply override the CSS style. E.g. by putting the following anywhere in one of your stylesheets: 

```css
.fixed-page-size {
  font-size: 26px; /** Any setting here - current default is 58px **/
}
```

The effect of overriding the font-size to 26px would look like this:

<img width="689" alt="Bildschirmfoto 2024-03-05 um 16 11 59" src="https://github.com/benwbrum/fromthepage/assets/470971/241080a6-0400-4426-976b-ce8556f476f8">

The more challenging question, however: how could make the overlay itself a bit smarter about auto-adjusting the font size? I guess we could try to come up with some kind of heuristic or simple formula that determines a font size based on the px height of the image and the number of lines?

**Comment by benwbrum** [comment]
I do like the `fixedPageSize` approach, and think that we should stick to that -- font adjustment should be manual after the text overlay is loaded, if we allow font adjustment at all.

Do we have the ability to determine the number of lines on the page, which we could use for calculating the font size?  As an alternative (possibly better), do we have the ability to determine the height of the average HTR line to match the font against?  As a third alternative, do we have any way of determining when text might overlap vertically by observing font sizes and baselines?

**Comment by rsimon** [comment]
> do we have the ability to determine the height of the average HTR line to match the font

Good point. Yes, the text overlay plugin loads the bounding box for each word on startup. It would be easy to determine the average over all of them.

I think the CSS `font-size` attribute won't perfectly match the measured bbox pixel height, so there might still be overlap, and it might be harder to compute that without digging too deeply into the DOM API. But just the average size might be a good start.

**Comment by benwbrum** [comment]
Closed by #3937
